pkg/instrumentors/bpf/github.com/gin-gonic/gin/probe.go

```diff
@@ -0,0 +1,217 @@
+	"errors"
+	"os"
+
+	"go.opentelemetry.io/auto/pkg/instrumentors/bpffs"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/link"
+	"github.com/cilium/ebpf/perf"
+	"go.opentelemetry.io/auto/pkg/inject"
+	"go.opentelemetry.io/auto/pkg/instrumentors/context"
+	"go.opentelemetry.io/auto/pkg/instrumentors/events"
+	"go.opentelemetry.io/auto/pkg/log"
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
+	"go.opentelemetry.io/otel/trace"
+	"golang.org/x/sys/unix"
+)
+
+//go:generate go run github.com/cilium/ebpf/cmd/bpf2go -target bpfel -cc clang -cflags $CFLAGS bpf ./bpf/probe.bpf.c
+
+type HttpEvent struct {
+	StartTime   uint64
+	EndTime     uint64
+	Method      [6]byte
+	Path        [100]byte
+	SpanContext context.EbpfSpanContext
+}
+
+type httpServerInstrumentor struct {
+	bpfObjects   *bpfObjects
+	uprobes      []link.Link
+	returnProbs  []link.Link
+	eventsReader *perf.Reader
+}
+
+func New() *httpServerInstrumentor {
+	return &httpServerInstrumentor{}
+}
+
+func (h *httpServerInstrumentor) LibraryName() string {
+	return "github.com/gin-gonic/gin"
+}
+
+func (h *httpServerInstrumentor) FuncNames() []string {
+	return []string{"github.com/gin-gonic/gin.(*Engine).ServeHTTP"}
+}
+
+func (h *httpServerInstrumentor) Load(ctx *context.InstrumentorContext) error {
+	spec, err := ctx.Injector.Inject(loadBpf, "go", ctx.TargetDetails.GoVersion.Original(), []*inject.InjectStructField{
+		{
+			VarName:    "method_ptr_pos",
+			StructName: "net/http.Request",
+			Field:      "Method",
+		},
+		{
+			VarName:    "url_ptr_pos",
+			StructName: "net/http.Request",
+			Field:      "URL",
+		},
+		{
+			VarName:    "path_ptr_pos",
+			StructName: "net/url.URL",
+			Field:      "Path",
+		},
+	}, false)
+
+	if err != nil {
+		return err
+	}
+
+	h.bpfObjects = &bpfObjects{}
+	err = spec.LoadAndAssign(h.bpfObjects, &ebpf.CollectionOptions{
+		Maps: ebpf.MapOptions{
+			PinPath: bpffs.BpfFsPath,
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, funcName := range h.FuncNames() {
+		h.registerProbes(ctx, funcName)
+	}
+
+	rd, err := perf.NewReader(h.bpfObjects.Events, os.Getpagesize())
+	if err != nil {
+		return err
+	}
+	h.eventsReader = rd
+
+	return nil
+}
+
+func (h *httpServerInstrumentor) registerProbes(ctx *context.InstrumentorContext, funcName string) {
+	logger := log.Logger.WithName("gin-gonic/gin-instrumentor").WithValues("function", funcName)
+	offset, err := ctx.TargetDetails.GetFunctionOffset(funcName)
+	if err != nil {
+		logger.Error(err, "could not find function start offset. Skipping")
+		return
+	}
+	retOffsets, err := ctx.TargetDetails.GetFunctionReturns(funcName)
+	if err != nil {
+		logger.Error(err, "could not find function end offsets. Skipping")
+		return
+	}
+
+	up, err := ctx.Executable.Uprobe("", h.bpfObjects.UprobeGinEngineServeHTTP, &link.UprobeOptions{
+		Address: offset,
+	})
+	if err != nil {
+		logger.V(1).Info("could not insert start uprobe. Skipping",
+			"error", err.Error())
```

Comment:
```suggestion
		logger.Error(err, "could not insert start uprobe. Skipping")
```

Reply:
I didn't do this at first because it matches what we're using in the other http probes and wanted to keep them mostly consistent. I'd like to apply the code styling fixes when we consolidate the http based probes in #101